File: WOT-iOS/WOTData/WOTData/WebServices/requests/Profile/WOTWEBRequestTankProfile.m
//
//  WOTWEBRequestTankProfile.m
//  WOT-iOS
//
//  Created on 9/8/15.
//  Copyright (c) 2015. All rights reserved.
//

#import "WOTWEBRequestTankProfile.h"
#import "WOTDataDefines.h"
#import <WOTData/WOTData-Swift.h>

@implementation WOTWEBRequestTankProfile

- (NSDictionary *)query {
    
    NSMutableDictionary *result = [[NSMutableDictionary alloc] init];
    result[WOTApiKeys.application_id] = [NSString valueOrSpaceString:self.hostConfiguration.applicationID];
    result[WOTApiKeys.fields] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.fields]];
    result[WOTApiKeys.tank_id] = [NSString valueOrSpaceString:[self.args escapedValueForKey:WOTApiKeys.tank_id]];
    return result;
}

- (NSString *)path {
    
    return @"/wot/encyclopedia/vehicleprofile/";
}


- (NSString *)method {
    
    return @"POST";
}

@end

File: WOT-iOS/WOTData/WOTData/WebServices/requests/WOTWEBRequestTankProfile.swift
//
//  WOTWEBRequestTankProfile.swift
//  WOTData
//
//  Created by Pavel Yeshchyk on 1/14/20.
//  Copyright © 2020 Pavel Yeshchyk. All rights reserved.
//

import Foundation

@objc
public class WOTWEBRequestTankProfile: WOTWEBRequest {
    
    override public var method: String { return "POST" }

    override public var path: String {
        return "/wot/encyclopedia/vehicleprofile/"
    }
}

extension WOTWEBRequestTankProfile: WOTModelServiceProtocol {
    
    @objc
    public class func modelClass() -> AnyClass? {
        return Vehicleprofile.self
    }
    
    @objc
    public func instanceModelClass() -> AnyClass? {
        return type(of: self).modelClass()
    }
}
